src/test/java/org/terasology/simpleFarming/systems/PlantAuthoritySystemTest.java

```diff
@@ -35,135 +23,98 @@
 import org.terasology.world.block.BlockManager;
 
 import java.util.Collections;
-import java.util.Set;
 
-import static org.junit.Assert.assertEquals;
 
-public class PlantAuthoritySystemTest extends ModuleTestingEnvironment {
+@ExtendWith(MTEExtension.class)
+@UseWorldGenerator("ModuleTestingEnvironment:empty")
+@Dependencies({"SimpleFarming", "CoreAssets", "ModuleTestingEnvironment"})
+public class PlantAuthoritySystemTest extends BaseAuthorityTest {
 
-    /** The block our test bush should use. */
+    /**
+     * The block our test bush should use.
+     */
     private static final String BUSH_BLOCK = "SimpleFarming:BaseBush";
 
-    private EntityManager entityManager;
-    private WorldProvider worldProvider;
+    @In
+    EntityManager entityManager;
+    @In
+    WorldProvider worldProvider;
+    @In
+    BlockManager blockManager;
+    @In
+    ModuleTestingHelper helper;
+
 
     private Block air;
     private Block dirt;
     private Block bush;
 
-    @Before
+    @BeforeEach
     public void initialize() {
-        entityManager = getHostContext().get(EntityManager.class);
-        worldProvider = getHostContext().get(WorldProvider.class);
-
-        BlockManager blockManager = getHostContext().get(BlockManager.class);
         air = blockManager.getBlock("engine:air");
-        dirt = blockManager.getBlock("CoreBlocks:Dirt");
+        dirt = blockManager.getBlock("CoreAssets:Dirt");
         bush = blockManager.getBlock(BUSH_BLOCK);
     }
 
-    @Override
-    public Set<String> getDependencies() {
-        return Collections.singleton("SimpleFarming");
-    }
-
     @Test
-    @Ignore // TODO Seems to enter infinite loop
     public void seedShouldGrowWherePlanted() {
-        plantSeedTest(dirt, air, true);
+        setBlock(new Vector3i(), dirt);
```

Comment:
I'd prefer to be explicit here in the test with the vector coordinates.
```suggestion
        setBlock(new Vector3i(0, 0, 0), dirt);
```